core/include/drivers/wdt.h

```diff
@@ -27,10 +31,99 @@ struct wdt_chip {
  * that control a watchdog device.
  */
 struct wdt_ops {
+	TEE_Result (*init)(struct wdt_chip *chip, unsigned long *min_timeout,
+			   unsigned long *max_timeout);
 	void (*start)(struct wdt_chip *chip);
 	void (*stop)(struct wdt_chip *chip);
 	void (*ping)(struct wdt_chip *chip);
 	TEE_Result (*set_timeout)(struct wdt_chip *chip, unsigned long timeout);
 };
 
+#ifdef CFG_WDT
+extern struct wdt_chip *wdt_chip;
+
+/* Register a watchdog as the system watchdog */
+TEE_Result watchdog_register(struct wdt_chip *chip);
+
+static inline
+TEE_Result watchdog_init(unsigned long *min_timeout, unsigned long *max_timeout)
+{
+	if (!wdt_chip)
+		return TEE_ERROR_NOT_SUPPORTED;
+
+	if (wdt_chip->ops->init)
```

Comment:
How about combining the check above with this to become
``` C
if (wdt_chip && wdt_chip->ops->init)
```
It saves a few lines looks better in my opinion. Same in for instance `watchdog_stop()` below.

Reply:
Acked, actually lack a `init` function should not be fatal and we should simply return `TEE_SUCCESS`